### PR TITLE
Symoplectic to Bargmann related code for unitaries

### PR DESCRIPTION
**Pull request by arsalan-motamedi**
**Context:**
Added a method ".symplectic" for the Unitary class that returns the symplectic representation of a Bargmann. Also, we can go back.
**Description of the Change:**
Described above.
**Benefits:**
Clear

### CODE REVIEW

File: mrmustard/lab_dev/transformations/base.py [hunk 1]
```diff
@@ -202,6 +203,35 @@ def inverse(self) -> Unitary:
             name=unitary_dual.name,
         )
 
+    @property
+    def symplectic(self):
+        r"""
+        Returns the symplectic representation
+        """
+        batch = self.representation.A.shape[0]
```
Comment (apchytr) [inline mrmustard/lab_dev/transformations/base.py:211]: This should simplify things
```suggestion
        batch_size = self.representation.ansatz.batch_size
        return [Au2Symplectic(self.representation.A[batch, :, :]) for batch in range(batch_size)]
```

File: mrmustard/lab_dev/states/base.py [hunk 2]
```diff
@@ -797,6 +797,22 @@ def __getitem__(self, modes: Union[int, Sequence[int]]) -> State:
...
+    #     """
+    #     m = len(modes)
+    #     max_idx = max(modes)
+
+    #     ancilla = list(range(max_idx + 1, max_idx + m + 1))
+    #     full_wires = modes + ancilla
+
+    #     psi = Ket.random(full_wires, max_r)
+    #     return psi[modes]
+
```
Comment (ziofil) [inline mrmustard/lab_dev/states/base.py:815]: remove if not needed

File: mrmustard/lab_dev/transformations/base.py [hunk 3]
```diff
@@ -202,6 +203,31 @@ def inverse(self) -> Unitary:
             name=unitary_dual.name,
         )
 
+    @property
+    def symplectic(self):
+        r"""
+        Returns the symplectic representation
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/base.py:209]:
```suggestion
        Returns the symplectic matrix that corresponds to this unitary
```

File: mrmustard/lab_dev/transformations/base.py [hunk 4]
```diff
@@ -202,6 +203,31 @@ def inverse(self) -> Unitary:
...
+        """
+        batch_size = self.representation.ansatz.batch_size
+        return [Au2Symplectic(self.representation.A[batch, :, :]) for batch in range(batch_size)]
+
+    @classmethod
+    def from_symplectic_directly(cls, modes, S) -> Unitary:
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/base.py:215]:
```suggestion
    def from_symplectic(cls, modes, S) -> Unitary:
```

File: mrmustard/lab_dev/transformations/base.py [hunk 5]
```diff
@@ -257,3 +283,18 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
...
+    #     m = len(wires)
+    #     S = math.random_symplectic(m, max_r)
+    #     modes = wires # replace everywhere
+    #     U = Unitary.random(modes = range(3*m))
+    #     vacuum_ancilla = Vacuum(range(2*m))
+    #     u_psi= vacuum_ancilla >> U
+    #     A = u_psi.representation # change name
+    #     B = u_psi.adjoint.representation
+    #     C = B[range(2*m)]@A[range(2*m)] # voila
+    #     return Channel.from_arguments(C, Wires(modes,modes,modes,modes))
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/base.py:300]: remove

File: tests/test_lab_dev/test_states.py [hunk 6]
```diff
@@ -358,6 +358,13 @@ def test_unsafe_batch_zipping(self):
         settings.UNSAFE_ZIP_BATCH = False
         assert better_cat == Coherent([0], x=2.0) + Coherent([0], x=-2.0)
 
+    # @pytest.mark.parametrize("max_sq", [1, 2, 3])
+    # def test_random_states(self, max_sq):
+    #     psi = Ket.random([1, 22], max_sq)
+    #     A = psi.representation.A[0]
+    #     assert np.isclose(psi.probability, 1)  # checks if the state is normalized
+    #     assert np.allclose(A - np.transpose(A), np.zeros(2))  # checks if the A matrix is symmetric
+
```
Comment (ziofil) [inline tests/test_lab_dev/test_states.py:367]: remove

File: tests/test_lab_dev/test_states.py [hunk 7]
```diff
@@ -583,6 +590,17 @@ def test_rshift(self):
...
+    # def test_random(self, modes):
+
+    #     m = len(modes)
+    #     dm = DM.random(modes)
+    #     A = dm.representation.A[0]
+    #     Gamma = A[:m, m:]
+    #     assert np.all(
+    #         np.linalg.eigvals(Gamma) >= 0
+    #     )  # checks if the off-diagonal block of dm is PSD
+
```
Comment (ziofil) [inline tests/test_lab_dev/test_states.py:603]: remove

**Review by ziofil — COMMENTED** [review]
I think you mixed up two branches (the random and this one). Just remove the commented code and it should be good to go!

**Pull request closed without merging**